perception/sub8_perception/nodes/dice_detect.py

```diff
@@ -0,0 +1,197 @@
+#!/usr/bin/env python
+'''
+script to detect dice , and mark the value of the dice
+
+Subscribes to Image topic
+Publishes Image and Points
+
+'''
+from __future__ import print_function
+import sys
+import rospy
+import math
+import cv2
+import numpy as np
+from sensor_msgs.msg import Image   # To publish Image
+from geometry_msgs.msg import Point     # To publish Points
+from cv_bridge import CvBridge, CvBridgeError
+import matplotlib.pyplot as plt
+import matplotlib.image as mpimg
+import pandas as pd
+from scipy.spatial import distance
+
+## Display Window size for Debug
+WINDOW_SIZE_H=1000
+WINDOW_SIZE_W=1000
+
+## Parameters for dice detections
+DIST_THRESHOLD=100
+SEARCH_RANGE_FOR_PIPS=2
+SENTINEL_INFINITE=10000
+
+class DiceDetect(object):
+    """docstring for DiceDetect."""
+    def __init__(self):
+        self.debug_image_publisher= rospy.Publisher("dice/debugimage", Image,queue_size=1)
+        self.point_publisher= rospy.Publisher("dice/points", Point, queue_size=1)
+        self.image_subscriber= rospy.Subscriber("/camera/seecam/image_color", Image, self.callback)
+        self.bridge = CvBridge()
+        ##self.dice_img = dice_img
+        ###self.detect()
+
+    def detect(self, dice_img):
+
+        # Setup SimpleBlobDetector parameters.
+        params = cv2.SimpleBlobDetector_Params()
+        params.filterByColor=1
+        params.blobColor= 0
+        params.minDistBetweenBlobs = 1
+        # Change thresholds
+        params.minThreshold = 0;
+        params.maxThreshold = 255;
+        # Filter by Area.
+        params.filterByArea = True
+        params.minArea = 10
+        # Filter by Circularity
+        params.filterByCircularity = True
+        params.minCircularity = 0.80
+        # Filter by Convexity
+        params.filterByConvexity = True
+        params.minConvexity = 0.87
+        # Filter by Inertia
+        params.filterByInertia = True
+        params.minInertiaRatio = 0.5
```

Comment:
I think it would be nice to have some correlation/debug comments so that we can quickly glance later on to see what values need to change